fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/metadata/expressions/KeyWithValueExpression.java

```diff
@@ -115,6 +115,8 @@ public KeyExpression getValueExpression() {
             List<KeyExpression> allKeys = normalizeKeyForPositions();
             if (splitPoint == allKeys.size()) {
                 valueExpression = EmptyKeyExpression.EMPTY;
+            } else if ((allKeys.size() - splitPoint) == 1) {
+                valueExpression = allKeys.get(splitPoint);
             } else {
                 valueExpression = new ThenKeyExpression(allKeys, splitPoint, allKeys.size());
             }
```

Comment:
Should this call `getInnerKey().getSubKey` like `getKeyExpression` does? I think in addition to single-column value bug added in the test, I think this also mishandles multi-column `KeyExpression`s in its children, which is what #458 was alluding to as well (though it was concerned with the key subexpression instead of the value subexpression). The `normalizeKeyForPositions` function _kind of_ handles it, but it doesn't get all of the nesteds and repeateds correct